support/android/apk/app/src/main/java/com/mozilla/servo/MainActivity.java

```diff
@@ -0,0 +1,303 @@
+        if (intent != null && intent.getAction().equals(Intent.ACTION_VIEW)) {
+            final String url = intent.getDataString();
+            if (url != null && URLUtil.isValidUrl(url)) {
+                Log.d(LOGTAG, "Received url "+url);
+                set_url(url);
+            }
+        }
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();  // Always call the superclass method first
+        Log.d(LOGTAG, "onStop");
+    }
+
+    @Override
+    protected void onPause() {
+        if (mWakeLock != null) {
+            mWakeLock.release();
+        }
+        super.onPause();
+        Log.d(LOGTAG, "onPause");
+    }
+
+    @Override
+    protected void onResume() {
+        if (mFullScreen) {
+            setFullScreen();
+        }
+        if (mWakeLock != null) {
+            mWakeLock.acquire();
+        }
+
+        super.onResume();
+        Log.d(LOGTAG, "onPause");
+    }
+
+    // keep the device's screen turned on and bright.
+    private void keepScreenOn() {
+        if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.JELLY_BEAN_MR1 ) {
+            getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+        }
+        else {
+            final PowerManager pm = (PowerManager) getSystemService(Context.POWER_SERVICE);
+            mWakeLock = pm.newWakeLock(PowerManager.SCREEN_DIM_WAKE_LOCK, getClass().getName());
+        }
+    }
+
+    // Dim toolbar and make the view fullscreen
+    private void setFullScreen()
+    {
+        if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            setFullScreenAndroid44();
+        }
+        else if( android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.ICE_CREAM_SANDWICH ) {
+            setFullScreenAndroid4();
+        }
+    }
+
+    @TargetApi(14)
+    private void setFullScreenAndroid4()
+    {
+        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_VISIBLE);
+        getWindow().getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LOW_PROFILE);
+    }
+
+    @TargetApi(19)
+    private void setFullScreenAndroid44()
+    {
+        View decorView = getWindow().getDecorView();
+        decorView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_STABLE
+                | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                | View.SYSTEM_UI_FLAG_FULLSCREEN
+                | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+    }
+
+    // This is here to make the toolbar autohide after 2 seconds of being touched
+    private void addFullScreenListener(final View view)
+    {
+        view.setOnSystemUiVisibilityChangeListener(
+                new View.OnSystemUiVisibilityChangeListener() {
+                    public void onSystemUiVisibilityChange(int visibility) {
+                        if (visibility == 0) {
```

Comment:
Please use the constants from https://developer.android.com/reference/android/view/View.OnSystemUiVisibilityChangeListener.html